### PR TITLE
Deprecate use of attributes to set viewModel properties

### PR DESCRIPTION
**Pull request by imaustink**
Related to #336

### CODE REVIEW

File: can-stache-bindings.js [hunk 1]
```diff
@@ -1202,9 +1208,20 @@ var decodeAttrName = function(name){
...
 		return;
 	}
+
+	if(
+		bindingInfo.child === "viewModel" &&
+		!isBindingsAttribute(bindingInfo.bindingAttributeName)
+	) {
+		var name = node.nodeName || node.name;
+		var value = node.nodeValue || node.value;
+		dev.warn(name + "=\"" + value + "\" is deprecated. Use " + name + ":from=\"'" + value + "'\" instead.");
```
Comment (phillipskevin) [inline can-stache-bindings.js]: Is it possible to add the filename / lineNumber to this warning?

File: can-stache-bindings.js [hunk 2]
```diff
@@ -1202,9 +1208,20 @@ var decodeAttrName = function(name){
...
+
 	if(!bindingInfo) {
 		return;
 	}
+
+	if(
+		bindingInfo.child === "viewModel" &&
+		!isBindingsAttribute(bindingInfo.bindingAttributeName)
+	) {
+		var name = node.nodeName || node.name;
```
Comment (phillipskevin) [inline can-stache-bindings.js:1221]: Do some Nodes not have `nodeName` and `nodeValue`? Are there tests for these different cases?

File: test/bindings-test.js [hunk 3]
```diff
@@ -3207,6 +3207,50 @@ testHelpers.dev.devOnlyTest('using @viewModel shows a deprecation warning', func
     equal(teardown(), 1);
 });
 
+test("set string on the viewModel", function(){
+	var ViewModel = DefaultMap.extend({
```
Comment (phillipskevin) [inline test/bindings-test.js]: You should add an `expect(2)` to this test.

File: test/bindings-test.js [hunk 4]
```diff
@@ -3207,6 +3207,50 @@ testHelpers.dev.devOnlyTest('using @viewModel shows a deprecation warning', func
...
+		tag: "test-elem",
+		viewModel: ViewModel
+	});
+
+	var template = stache("<test-elem foo:from=\"'bar'\" baz:from=\"'qux'\"/>");
+	template();
+});
+
+test("warn about using attributes to set values on the viewModel", function(){
+	var teardown = testHelpers.dev.willWarn(/foo="bar" is deprecated\. Use foo:from="'bar'" instead\./);
```
Comment (phillipskevin) [inline test/bindings-test.js]: This tests seems to imply that only one of these attributes gives a warning, when really they both should. Can you update the regex to match both cases?

**Review by phillipskevin — CHANGES_REQUESTED** [review]

File: can-stache-bindings.js (hunk 2 above)
Comment (imaustink) [reply can-stache-bindings.js:1221]: In `VDOM`, nodes have `name` and `value` instead. Since the tests run in both `DOM` and `VDOM`, this is tested.

**Comment by imaustink** [comment]
@phillipskevin I've updated the PR to address your concerns. For the tests to pass, I need to release these changes: https://github.com/canjs/can-stache/pull/373.

**Comment by imaustink** [comment]
@phillipskevin this should be ready to go if you want to look it over one more time.

File: can-stache-bindings.js [hunk 5]
```diff
@@ -1202,9 +1208,29 @@ var decodeAttrName = function(name){
...
+		var value = node.nodeValue || node.value;
+		var filename = bindingData.scope.peek("scope.filename");
+		var lineNumber = bindingData.scope.peek('scope.lineNumber');
+
+		dev.warn(
+			(filename ? filename + ": " : "") +
```
Comment (phillipskevin) [inline can-stache-bindings.js]: Sorry to nitpick, but I think these warnings look weird when there is a space before the line number. Can you update it so it looks like
`file.stache:1: foo=...`?

**Review by phillipskevin — CHANGES_REQUESTED** [review]
Just one tiny thing. Feel free to dismiss my review and merge/release once that is changed.

**Pull request merged**